source/crud/tests/unified/bulkWrite-arrayFilters.yml

```diff
@@ -0,0 +1,180 @@
+# This file was created automatically using mongodb-spec-converter.
+# Please review the generated file, then remove this notice.
+
+description: bulkWrite-arrayFilters
+schemaVersion: '1.1'
+createEntities:
+  -
+    client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      uriOptions:
+        retryReads: false
+      observeEvents:
+        - commandStartedEvent
+  -
+    database:
+      id: &database0 database0
+      client: client0
+      databaseName: &database_name crud-tests
+  -
+    collection:
+      id: &collection0 collection0
+      database: database0
+      collectionName: &collection_name test
+runOnRequirements:
+  -
+    minServerVersion: 3.5.6
+initialData:
+  -
+    collectionName: *collection_name
+    databaseName: *database_name
+    documents:
+      -
+        _id: 1
+        'y':
+          -
+            b: 3
+          -
+            b: 1
+      -
+        _id: 2
+        'y':
+          -
+            b: 0
+          -
+            b: 1
+tests:
+  -
+    description: 'BulkWrite updateOne with arrayFilters'
+    operations:
+      -
+        object: *collection0
+        name: bulkWrite
+        arguments:
+          requests:
+            -
+              updateOne:
+                filter: {  }
+                update:
+                  $set:
+                    'y.$[i].b': 2
```

Comment:
Did the dollar or brackets cause the spec converter to quote this field, or is that due to internal logic in Symfony's YAML library? If this is something you have control over, I'd suggest maybe just applying the quoting if the field contains _any_ special characters. That'd probably be consistent with most of our hand-written files.

Reply:
There's no control over this, and the library seems to only quote if the string contains problematic characters.